locust/contrib/geventhttpclient.py

```diff
@@ -61,23 +66,31 @@ class GeventHttpSession(object):
     def __init__(self, base_url):
         self.base_url = base_url
         self.cookiejar = CookieJar()
-        self.client = UserAgent(max_retries=1, cookiejar=self.cookiejar)
-        
+        self.client = LocustUserAgent(max_retries=1, cookiejar=self.cookiejar)
+    
+    def _build_url(self, path):
+        """ prepend url with hostname unless it's already an absolute URL """
+        if absolute_http_url_regexp.match(path):
+            return path
+        else:
+            return "%s%s" % (self.base_url, path)
+    
     def request(self, method, path, name=None, **kwargs):
-        url = self.base_url + path
+        # prepend url with hostname unless it's already an absolute URL
+        url = self._build_url(path)
         
```

Comment:
you can probably simplify or remove the `_build_url()` method and just use something like:

`url = path if path.startswith('http') else urljoin(base_url, path)`